### PR TITLE
feat: add LLA distance calculation

### PR DESCRIPTION
**Pull request by lucas-bremond**

### CODE REVIEW

File: src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp [hunk 1]
```diff
@@ -99,6 +101,13 @@ Length LLA::getAltitude() const
     return altitude_;
 }
 
+Length LLA::calculateDistanceTo(
+    const LLA& aLLA, const Length& anEllipsoidEquatorialRadius, const Real& anEllipsoidFlattening
```
Comment (lucas-bremond) [inline src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp:105]: Could we make clang format put each arg on its own line (especially when adding a trailing `,` after the last one, like black does?

Comment (vishwa2710) [reply src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp:105]: Unfortunately this is not possible AFAIK: https://stackoverflow.com/questions/50660455/clang-format-always-break-all-parameters-one-per-line

:( Hopefully they add this capability in the future.

Comment (vishwa2710) [reply src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp:105]: The trailing comma works for when using bracket initialization:
```
return {
        Derived::Undefined(),
        Length::Undefined(),
        Real::Undefined(),
        Real::Undefined(),
        Real::Undefined(),
    };
```
But not for function arguments.

Comment (lucas-bremond) [reply src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp:105]: Ok!

**Review by Derollez — APPROVED** [review]

File: bindings/python/test/coordinate/spherical/test_lla.py [hunk 2]
```diff
@@ -10,146 +10,287 @@
...
+WGS84_EGM96 = physics.environment.objects.celestial_bodies.earth.models.WGS84_EGM96
 
 
-def test_coordinate_spherical_lla_constructors():
-    # Main Constructor
-    latitude: Angle = Angle(30.0, Angle.Unit.Degree)
-    longitude: Angle = Angle(30.0, Angle.Unit.Degree)
-    altitude: Length = Length(1000.0, Length.Unit.Meter)
+@pytest.fixture
+def latitude_float() -> float:
```
Comment (lucas-bremond) [inline bindings/python/test/coordinate/spherical/test_lla.py:18]: Proposed improved naming:

```suggestion
def latitude_deg() -> float:
```

File: bindings/python/test/coordinate/spherical/test_lla.py [hunk 3]
```diff
@@ -10,146 +10,287 @@
...
+    return LLA(
+        Angle(-90.0, Angle.Unit.Degree),
+        Angle(0.0, Angle.Unit.Degree),
+        Length(1.0, Length.Unit.Meter),
+    )
+
+
+class TestLLA:
+    def test_constructor_list_success(
+        self, latitude_float: float, longitude_float: float, altitude_float: float
```
Comment (lucas-bremond) [inline bindings/python/test/coordinate/spherical/test_lla.py:99]: Add trailing `,` to make black pretty

```suggestion
        self,
        latitude_float: float,
        longitude_float: float,
        altitude_float: float,
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.test.cpp [hunk 4]
```diff
@@ -347,3 +347,126 @@ TEST_F(OpenSpaceToolkit_Physics_Coordinate_Spherical_LLA, Cartesian)
         EXPECT_ANY_THROW(LLA::Cartesian(Vector3d::Undefined(), Length::Undefined(), Real::Undefined()));
     }
 }
+
+TEST_F(OpenSpaceToolkit_Physics_Coordinate_Spherical_LLA, DistanceBetween)
```
Comment (lucas-bremond) [inline test/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.test.cpp:351]: Shall we add a test for `distanceTo` as well?

Comment (lucas-bremond) [reply test/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.test.cpp:351]: I'd also recommend splitting into two test `DistanceBetween_Spherical` and `DistanceBetween_WGS84`

Comment (Derollez) [reply test/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.test.cpp:351]: > Shall we add a test for distanceTo as well?

Oups, missed that! Will do

**Review by Derollez — APPROVED** [review]

**Pull request merged**